### ISSUE TITLE
Port twisted.spread.pb to Python3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7598]
|<img alt="wolfgang61's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| wolfgang61 reported|
|-|-|
|Trac ID|trac#7598|
|Type|enhancement|
|Created|2014-08-29 02:59:39Z|
|Branch|https://github.com/twisted/twisted/tree/port-pb-7598|

This includes not yet ported lower level modules. Found so far:

banana
styles

<details><summary>Searchable metadata</summary>

```
trac-id__7598 7598
type__enhancement enhancement
reporter__wolfgang61 wolfgang61
priority__normal normal
milestone__Python_3_x Python-3.x
branch__port_pb_7598 port-pb-7598
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1409281179315772 1409281179315772
changetime__1472068444945208 1472068444945208
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

Opening up a PR to start seeing CI results presented: https://github.com/twisted/twisted/pull/436

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph removed owner|
|-|-|

The branch there ports PB to Python 3. There might still be a few coverage issues, and since I trust codecov, I'm not entiirely sure.  But this is definitely worth at least a first pass review at this point.

It also significantly refactors a lot of code duplication especially in the area of deserialization; previously there were about six different things that needed to "make a new object" and would do them in various subtly broken ways.  In this port branch it is now narrowed down to one, along with some other benefits (the weird lifecycle-management hack that Cacheable uses is now visibly _part_ of RemoteCache rather than just floating around inside jelly).

I tried to keep it as minimal as possible but it's still about 2x as big as I've been looking to make patches these days.  If it's really tough for a reviewer to get their head around this I could probably do, say, just the Jelly changes first, but hopefully it's comprehensible enough that we won't need to resort to breaking it up.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @glyph|
|-|-|

https://github.com/twisted/twisted/pull/436#issuecomment-241970992

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set status to `closed`|
|-|-|

In changeset 7133de6b678c101e3a62119801ce0c7579f0f7d2
```
#!CommitTicketReference repository="" revision="7133de6b678c101e3a62119801ce0c7579f0f7d2"
Merge pull request #436 from twisted/port-pb-7598

Author: hawkowl, glyph

Reviewer: rodrigc

Fixes: #7598 

Port Perspective Broker to Python 3.
```
